third-party/realdds/src/dds-stream-profile.cpp

```diff
@@ -0,0 +1,168 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#include <realdds/dds-stream-profile.h>
+#include <realdds/dds-exceptions.h>
+#include <map>
+
+
+namespace realdds {
+
+
+std::string dds_stream_uid::to_string() const
+{
+    std::ostringstream os;
+    os << "0x" << std::hex << whole;
+    return os.str();
+}
+
+
+dds_stream_format::dds_stream_format( std::string const & s )
+{
+    if( s.length() > size )
+        DDS_THROW( runtime_error, "format is too long" );
+    data[s.copy( data, size )] = 0;
+}
+
+
+template< typename T > uint32_t rs_fourcc( const T a, const T b, const T c, const T d )
+{
+    static_assert( ( std::is_integral< T >::value ), "rs_fourcc supports integral built-in types only" );
+    return ( ( static_cast< uint32_t >( a ) << 24 ) | ( static_cast< uint32_t >( b ) << 16 )
+             | ( static_cast< uint32_t >( c ) << 8 ) | ( static_cast< uint32_t >( d ) << 0 ) );
+}
+
+
+enum rs2_format  // copy from rs2_sensor.h

```

Comment:
The rs2 enum and conversion functions will need to be synchronized to LibRS all the time, for example adding the HKR types. Isn't casting to int enough?

Reply:
The conversion functions and decl are temporary -- will be removed when the notification/control mechanism is not int-based.